source/elements/oneDPL/source/parallel_api.rst

```diff
@@ -69,10 +105,14 @@ device_policy class
   };
 
 An object of the ``device_policy`` type is associated with a ``sycl::queue`` that is used
-to run algorithms on a DPC++ compliant device.
+to run algorithms on a DPC++ compliant device. When an algorithm runs with ``device_policy``
+it is capable of processing SYCL buffers (passed via :code:`oneapi::dpl::begin/end`),
+data in the host memory or data in Unified Shared Memory (USM), inlcuding USM device memory.
+When data is placed in the host memory, it shall be passed to the algorithms as random access
+iterators. oneDPL algorithm might do implicit transfering data to a device before execution.
```

Comment:
I would use something like the following:

> Data placed in the host memory and USM can only be passed to oneDPL algorithms as pointers and random access iterators. The way to transfer data from the host memory to a device and back is unspecified; per-element data movement to/from a temporary storage is a possible valid implementation.

The wording is discussable, but the two key ideas are:
- data in the host memory and in USM can be passed to algorithms in the same ways
- an implementation may, if needed, implicitly copy data from host memory to a temporary storage and back

Reply:
done